symengine/add.h

```diff
@@ -69,11 +69,11 @@ class Add : public Basic
 
     virtual vec_basic get_args() const;
 
-    RCP<const Number> get_coef() const
+    inline const RCP<const Number> &get_coef() const
     {
         return coef_;
     }
```

Comment:
So the only issue I have with this PR are changes like these. A safe rule is to never return `&` from a function, because it's too easy to do something like:
```c++
// create `e`
const auto a = e.get_coef();
// destroy `e`

std::cout << *a; // UNDEFINED
```
I'll write soon about some workarounds for this.